# Trabajo Final

## Kozume Tsuchida y Nicolás Gutiérrez

## Para este trabajo, vamos a analizar la base de datos pertinente a los 100 ships más populares para el Semestre 1 del 2024. Para esto, haremos uso de la tabla elaborada por el usuario "centreoftheselights" (disponible en https://archiveofourown.org/works/57864097/chapters/147284449). Esta contiene: la posición (Rank), el cambio relativo que ha tenido con respecto a la edición anterior de la base de datos (Change), la relación sobre la que se ha escrito, también conocida como "ship" (Relationship), el universo temático al cual pertenece esta relación ("Fandom"), la cantidad de obras escrita en el periodo de enero a julio del 2024 (Works Gained), el total de obras hasta la fecha (Total Works), el tipo de relación (Type), y la etnicidad de los participantes (Race1 y Race2). 

### Preparar la base de datos

In [2]:
import pandas as pd 
import altair as alt

data = pd.read_excel("C:/data/AO3_Ship_Stats_Summer_2024.xlsx")

In [6]:
print(data)

   Rank Change                                       Relationship  \
0     1     18                    Aziraphale/Crowley (Good Omens)   
1     2      0                           Sirius Black/Remus Lupin   
2     3      1    Dazai Osamu/Nakahara Chuuya (Bungou Stray Dogs)   
3     4    New                       Astarion/Tav (Baldur's Gate)   
4     5     43                           Getou Suguru/Gojo Satoru   
..  ...    ...                                                ...   
95  95=    New                       Rafael Lange | Cellbit/Roier   
96   97    -72  Aemond "One-Eye" Targaryen/Lucerys Velaryon (S...   
97   98    -24                       Aoyagi Touya/Shinonome Akito   
98   99    New                            Ivan/Till (Alien Stage)   
99  100    -76             Vash the Stampede/Nicholas D. Wolfwood   

                                        Fandom  Works Gained  Total Works  \
0                              Good Omens (TV)         14770        59648   
1                

### ¿Qué "ships" han crecido más durante el primer semestre de 2024?

In [3]:
data["crecimiento"] = (data["Works Gained"] / data["Total Works"]) * 100

In [4]:
subset = data[["Relationship", "Fandom", "crecimiento"]]

In [5]:
data[["Relationship", "Fandom", "crecimiento"]].sort_values(by="crecimiento")

,Relationship,Fandom,crecimiento
37,Castiel/Dean Winchester,Supernatural (TV 2005),2.432540
58,Adrien Agreste | Chat Noir/Marinette Dupain-Ch...,Miraculous Ladybug,5.667578
76,Jeon Jungkook/Park Jimin,Bangtan Boys | BTS,5.910736
73,Merlin/Arthur Pendragon (Merlin),Merlin (TV),5.953705
42,Jeon Jungkook/Kim Taehyung | V,Bangtan Boys | BTS,6.277607
...,...,...,...
44,Astarion/Gale (Baldur's Gate),Baldur's Gate (Video Games),100.000000
55,Astarion/The Dark Urge (Baldur's Gate),Baldur's Gate (Video Games),100.000000
48,Alastor/Vox (Hazbin Hotel),Hazbin Hotel (Cartoon),100.000000
10,Neuvillette/Wriothesley (Genshin Impact),原神 | Genshin Impact (Video Game),100.000000


#### Como se puede observar, 11 de los 100 ships han experimentado un crecimiento del 100%. Es decir, todas las obras hasta el momento fueron publicadas dentro del primer semestre del 2024. El ship "Angel Dust/Husk" creció en 86.37%, mientras que "Blade/Dan Heng" creció en 84.6%

#### Los ships que menos crecieron fueron "Castiel/Dean Winchester" y "Cat-Noir/Ladybug", con 2.45% y 5.67%, respectivamente. Esto puede deberse a que se basan en material que existe desde mucho antes del 2024, por lo que ya hay un "corpus" de obras más completo que en los otros casos. 

In [6]:
alt.Chart(data).mark_bar().encode(
    x=alt.X("Relationship", sort="y"),  
    y=alt.Y("crecimiento"))

alt.Chart(...)

### ¿Qué fandoms tienen más obras totales?

In [7]:
fandom = data.groupby("Fandom")["Total Works"].sum()

print(fandom.sort_values())

Fandom
Alien Stage (Web Series)                                          1398
QSMP | Quackity SMP                                               1812
ZEROBASEONE | ZB1 (Korea Band)                                    2641
Slam Dunk (Anime & Manga)                                         2650
Blue Lock (Manga)                                                 3410
Formula 1 RPF                                                     3636
Hermitcraft SMP                                                   4099
Trigun (Anime & Manga 1995-2008)                                  4377
Helluva Boss (Web Series)                                         4632
Loki (TV 2021)                                                    4897
Bridgerton (TV)                                                   5034
Rise of the Teenage Mutant Ninja Turtles (Cartoon 2018)           5098
呪術廻戦 | Jujutsu Kaisen (Anime)                                     5435
SEVENTEEN (Band)                                                  6250

#### Según los resultados, vemos que los ships pertenecientes a los fandoms de Harry Potter, Supernatural, My Hero Academia, Good Omen y BTS son los que tienen la mayor cantidad de obras totales. Asimismo, Alien Stage y Quackity SMP son los fandoms con menor cantidad de obras totales. Esto puede deberse a que se trata de contenido más reciente o más nicho, mientras que los otros responden a series o películas "mainstream", por lo que disfrutan de una mayor audiencia

### ¿Sobre qué etnicidades se escribe más en AO3?

In [8]:
def calcular_etnicidad(row):
    return row["Race1"] if row["Race1"] == row["Race2"] else "Mixed"

In [9]:
data["ship_etnicidad"] = data.apply(calcular_etnicidad, axis=1)

In [10]:
etni = data.groupby("ship_etnicidad")["Total Works"].sum()
print(etni)

ship_etnicidad
Ambig       5775
Asian     396265
Latino      1812
MENA       11947
Mixed     260987
N.H.       29482
White     618179
Name: Total Works, dtype: int64


#### De acuerdo a los resultados, se ha escrito principalmente sobre "ships" entre personajes blancos en primer lugar, asiáticos en segundo. En tercer lugar, se encuentran los "ships" hechos con personajes de etnicidades distintas entre si. Los "ships" con menor representación son entre personajes latinos.


In [11]:
etni = etni.reset_index()

alt.Chart(etni).mark_bar().encode(
    x="Total Works",
    y="ship_etnicidad"
)

alt.Chart(...)

### ¿Cuáles son las preferencias narrativas según la diversidad de ships?

In [12]:
ship = data.groupby("Type")["Total Works"].sum()
print(ship)

Type
F/F       41705
F/M      168843
Gen      128532
M/M      898780
Other     86587
Name: Total Works, dtype: int64


In [13]:
ship = ship.reset_index()

alt.Chart(ship).mark_bar().encode(
    x=alt.X("Type", sort="y"),  
    y=alt.Y("Total Works"),
    color=alt.Color("Type"))

alt.Chart(...)

**El tipo de relación M/M (masculino/masculino) tiene la mayor cantidad de ships, lo que indica una fuerte preferencia o interés de la comunidad por este tipo de representación en las historias. F/M (femenino/masculino) ocupa el segundo lugar, reflejando una representación significativa de relaciones heterosexuales. Relativamente pocas historias se centran en F/F (femenino/femenino), Gen (relaciones no románticas) y Other (otras categorías). Esto sugiere que estos tipos tienen menor popularidad o atención.**